addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusBaseThingHandler.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fronius.handler;
+
+import java.math.BigDecimal;
+import java.util.Calendar;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.fronius.FroniusBridgeConfiguration;
+import org.openhab.binding.fronius.internal.api.ValueUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Basic Handler class for all Fronius services.
+ *
+ * @author Gerrit Beine - Initial contribution
+ * @author Thomas Rokohl - Refactoring to merge the concepts
+ */
+public abstract class FroniusBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusBaseThingHandler.class);
+    private final String serviceDescription;
+    private FroniusBridgeHandler bridgeHandler;
+
+    public FroniusBaseThingHandler(Thing thing) {
+        super(thing);
+        serviceDescription = getDescription();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            updateChannels();

```

Comment:
This might be a bit too simple - you should actually only update the one that has been requested to be refreshed here.
Also note that you might receive a REFRESH command for every channel at startup - in order to avoid too many requests to the inverter, you might want to use some caching, see https://www.eclipse.org/smarthome/documentation/features/frameworkUtilities.html#caching

Reply:
i update the concrete channel now.

caching comes in a later version